File: app/src/main/java/io/github/benoitduffez/cupsprint/detect/PrinterResult.java
package io.github.benoitduffez.cupsprint.detect;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class PrinterResult {
    private List<PrinterRec> printerRecs;

    private List<String> errors;

    PrinterResult() {
        printerRecs = Collections.synchronizedList(new ArrayList<PrinterRec>());
        errors = Collections.synchronizedList(new ArrayList<String>());
    }

    public List<PrinterRec> getPrinters() {
        return printerRecs;
    }

    public List<String> getErrors() {
        return errors;
    }

    void setPrinterRecs(ArrayList<PrinterRec> printerRecs) {
        this.printerRecs = printerRecs;
    }
}

File: app/src/main/java/io/github/benoitduffez/cupsprint/detect/PrinterResult.kt
package io.github.benoitduffez.cupsprint.detect

import java.util.*

class PrinterResult internal constructor() {
    var printers: List<PrinterRec>? = null
        private set

    private val errors: List<String>

    init {
        printers = Collections.synchronizedList(ArrayList())
        errors = Collections.synchronizedList(ArrayList())
    }

    internal fun setPrinterRecs(printerRecs: ArrayList<PrinterRec>) {
        printers = printerRecs
    }
}
